src/Angor/Client/Components/MessageComponent.razor

```diff
@@ -0,0 +1,637 @@
+                    bool isFromUserToContact = _.Event.Pubkey == currentUserPubKey && 
+                                              tagValue == ContactHexPub;
+                                              
+                    bool isFromContactToUser = _.Event.Pubkey == ContactHexPub && 
+                                              tagValue == currentUserPubKey;
+                    
+                    return isFromUserToContact || isFromContactToUser;
+                })
+                .Subscribe(async messageEvent => {
+                    await ProcessDirectMessage(messageEvent.Event);
+                    StateHasChanged();
+                });
+        }
+        catch (Exception ex) {
+            Logger.LogError($"Error in SubscribeToMessages: {ex.Message}");
+        }
+    }
+
+    // Process and deduplicate messages
+    private async Task ProcessDirectMessage(NostrEvent eventMessage)
+    {
+        try
+        {
+            if (eventMessage == null || string.IsNullOrEmpty(eventMessage.Content)) {
+                return;
+            }
+
+            // Skip already processed messages with the same ID
+            if (directMessages.Any(m => m.Id == eventMessage.Id)) {
+                return;
+            }
+
+            // Check if message with same content already exists
+            string currentUserPubKey = NostrHelper.ConvertBech32ToHex(CurrentUserNpub);
+            bool isFromCurrentUser = eventMessage.Pubkey == currentUserPubKey;
+            
+            // Decrypt the message
+            string decryptedContent;
+            try
+            {
+                // Get the partner's pubkey
+                string partnerPubKeyHex = isFromCurrentUser
+                    ? eventMessage.Tags?.FindFirstTagValue(NostrEventTag.ProfileIdentifier)
+                    : eventMessage.Pubkey;
+                    
+                decryptedContent = await _encryptionService.DecryptNostrContentAsync(
+                    CurrentUserPrivateKeyHex,
+                    partnerPubKeyHex,
+                    eventMessage.Content
+                );
+            }
+            catch (Exception)
+            {
+                try
+                {
+                    // Try alternative format
+                    string alternativeKey = isFromCurrentUser
+                        ? ContactHexPub
+                        : currentUserPubKey;
+                        
+                    decryptedContent = await _encryptionService.DecryptNostrContentAsync(
+                        CurrentUserPrivateKeyHex,
+                        alternativeKey,
+                        eventMessage.Content
+                    );
+                }
+                catch
+                {
+                    decryptedContent = "[Could not decrypt message]";
+                }
+            }
+
+            // Check for duplicate message content with same sender
+            if (directMessages.Any(m => 
```

Comment:
Why would we have duplicates? are you by any chance receiving form multiple relays the same messages?